Pylint 3.3.0 does not recognize f-string as string and reports no-member for startswith()

### Bug description

The following file test_pylint.py:

```python
#!/usr/bin/env python
"""module"""


class Cls:
    # pylint: disable=too-few-public-methods
    """Cls"""


c_obj = Cls()

c_repr1 = '{!r}'.format(c_obj)  # pylint: disable=consider-using-f-string
c_repr2 = f'{c_obj!r}'

assert c_repr1.startswith('<__main__.Cls')
assert c_repr2.startswith('<__main__.Cls')  # pylint reports no-member
```

causes pylint to report this issue:

```
$ pylint test_pylint.py 
************* Module tmp_issues.test_pylint
test_pylint.py:16:7: E1101: Instance of 'Cls' has no 'startswith' member (no-member)
```

I tested this with multiple pylint versions, and the first version that has this behavior is pylint 3.3.0.

### Command used

```shell
pylint test_pylint.py
```


### Pylint output

```python
************* Module test_pylint
test_pylint.py:16:7: E1101: Instance of 'Cls' has no 'startswith' member (no-member)
```


### Expected behavior

Pylint should not report this as an error, because an f-string is a string and has the startswith() method.

### Pylint version

```shell
pylint    3.3.0
astroid   3.3.3
Python 3.12.4 (main, Jun  6 2024, 18:26:44) [Clang 15.0.0 (clang-1500.3.9.4)]
```


### OS / Environment

macOS Sonoma 14.6.1

--- a/doc/whatsnew/fragments/9947.false_positive
+++ b/doc/whatsnew/fragments/9947.false_positive
@@ -1,0 +1,3 @@
+Fix regression causing some f-strings to not be inferred as strings.
+
+Closes #9947

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -41,7 +41,7 @@ dependencies    = [
     # Also upgrade requirements_test_min.txt.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/pylint-dev/astroid/issues/1341
-    "astroid>=3.3.3,<=3.4.0-dev0",
+    "astroid>=3.3.4,<=3.4.0-dev0",
     "isort>=4.2.5,<6,!=5.13.0",
     "mccabe>=0.6,<0.8",
     "tomli>=1.1.0;python_version<'3.11'",

--- a/requirements_test_min.txt
+++ b/requirements_test_min.txt
@@ -1,6 +1,6 @@
 .[testutils,spelling]
 # astroid dependency is also defined in pyproject.toml
-astroid==3.3.3  # Pinned to a specific version for tests
+astroid==3.3.4  # Pinned to a specific version for tests
 typing-extensions~=4.12
 py~=1.11.0
 pytest~=8.3

